### ISSUE TITLE
Morph model deformations

### ISSUE THREAD

**Issue by zicodxx** [dxx-rebirth/dxx-rebirth#277]
I observed this issue in D1X-Rebirth. D2X-Rebirth untested for it. Bisect didn't happen, yet either.
When a bot comes out of a materialization center, some of it's textures are deformed. I saw this on their visors. Also parts that should be black, such as gun barrels or engines render as white. The following screenshot shows a bot with both issues visible: 
http://www.dxx-rebirth.com/tmp/scrn0021.jpg

EDIT: I should mention that as soon as the bots are fully "grown", their graphical glitches disappear.

UPDATE 1: This seems to be a pretty long-standing regression. I could observe this issue in 0.58.1 (UPDATE2: even as far back as 0.57) as well. Software rendering shows the issue (both in fact) as well so it does not seem to be related to OpenGL rendering. I can confirm however that both issues do not occur in the original MS-DOS release which I played with the exact same game files (to make sure it's not a model issue within descent.pig)

The bug is easy to reproduce in Descent 1, level 5: From start of the level, fly tho the set of doors. Open the left one. To your left side there is a blue hallway. At the end of it, fly right. You should find the blue key and a materialization center. This one produces Laser Hulks and Lifters. Especially the former show the issue very well. On the Lifters, you may be able to spot their back which shows the engine rendered in white.

UPDATE 3: The static white color on flat poly's of morphing models goes back to a change in d1x-rebirth 0.57 in interp.c. Instead to read the color from the model (w(p+28)), the color was set to 55. The changelogs of that time are fairly unclear on why I did that but I assume it was a mistake on my end - a test-change I forgot to revert when I worked on supporting the GP2X platform back in the day. I'll revert this change and commit as soon as I know more about the animated texture issues of the morphing models. That problem seems to be separate and goes even further back. 
The code in question was initially written in assembler and I start to wonder if its conversion to C introduced this issue.

**Comment by zicodxx** [comment]
Part 1 of the issue - white flat polys on morph models - is now fixed in 34b24c0d69a5011a3c9ccece16975d848032d0a2

**Comment by vLKp** [comment]
I always wondered why `55` in one game and not the other.

**Comment by zicodxx** [comment]
Pushed daad12106d5e247b01896be6edc878621f2590e6 which should fix the issue with the model textures. I wasn't able to trace the issue back to its origin. The very first C version of interp.c I could find in D1X-1.43 already shows this issue which makes me assume it was an oversight when the Assembler code was converted to C (which was't the first time: 47c00c7128d101d701f849322a79222bef7ffdea). 
Unfortunately that made me unable to properly restore the code into its original behavior as my skills reading Assembler are rather low. Instead I used the instructions for op_tmappoly from g3_draw_polygon_model_state() which properly fixed this bug. 

I wrote a small debugging key combo which  would generate all possible robots in their morphing state one after another. With this I was able to verify the morphing state of all robots and from what I can tell, it looks correct now. I have kept the original code for op_tmappoly by just commenting it out - for further reference. I know we could just go back to an older revision but it's just a few extra lines and may make the changes a bit more clear if we ever need to revisit this (which I hope we won't need to). I also added the debug keys (24bd4170f42f4acec01a6fcb2e2cd097603a4016) to generate morphing robots. Since those result in normal robots, this can come in handy for all sorts of robot testing.
